druid-shell/src/platform/mac/keycodes.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! macOS keycode handling.
+
+use crate::keycodes::KeyCode;
+
+pub type RawKeyCode = u16;
+
+impl From<u16> for KeyCode {
+    fn from(raw: u16) -> KeyCode {
+        match raw {
+            0x00 => KeyCode::KeyA,
+            0x01 => KeyCode::KeyS,
+            0x02 => KeyCode::KeyD,
+            0x03 => KeyCode::KeyF,
+            0x04 => KeyCode::KeyH,
+            0x05 => KeyCode::KeyG,
+            0x06 => KeyCode::KeyZ,
+            0x07 => KeyCode::KeyX,
+            0x08 => KeyCode::KeyC,
+            0x09 => KeyCode::KeyV,
+            //0x0a => World 1,
+            0x0b => KeyCode::KeyB,
+            0x0c => KeyCode::KeyQ,
+            0x0d => KeyCode::KeyW,
+            0x0e => KeyCode::KeyE,
+            0x0f => KeyCode::KeyR,
+            0x10 => KeyCode::KeyY,
+            0x11 => KeyCode::KeyT,
+            0x12 => KeyCode::Key1,
+            0x13 => KeyCode::Key2,
+            0x14 => KeyCode::Key3,
+            0x15 => KeyCode::Key4,
+            0x16 => KeyCode::Key6,
+            0x17 => KeyCode::Key5,
+            0x18 => KeyCode::Equals,
+            0x19 => KeyCode::Key9,
+            0x1a => KeyCode::Key7,
+            0x1b => KeyCode::Minus,
+            0x1c => KeyCode::Key8,
+            0x1d => KeyCode::Key0,
+            0x1e => KeyCode::RightBracket,
+            0x1f => KeyCode::KeyO,
+            0x20 => KeyCode::KeyU,
+            0x21 => KeyCode::LeftBracket,
+            0x22 => KeyCode::KeyI,
+            0x23 => KeyCode::KeyP,
+            0x24 => KeyCode::Return,
+            0x25 => KeyCode::KeyL,
+            0x26 => KeyCode::KeyJ,
+            0x27 => KeyCode::Backtick,
+            0x28 => KeyCode::KeyK,
+            0x29 => KeyCode::Semicolon,
+            0x2a => KeyCode::Backslash,
+            0x2b => KeyCode::Comma,
+            0x2c => KeyCode::Slash,
+            0x2d => KeyCode::KeyN,
+            0x2e => KeyCode::KeyM,
+            0x2f => KeyCode::Period,
+            0x30 => KeyCode::Tab,
+            0x31 => KeyCode::Space,
+            0x32 => KeyCode::Backtick,
+            0x33 => KeyCode::Backspace,
+            //0x34 => unkown,
```

Comment:
These should be "unknown". I know that's pre-existing, but as long as the diff is so large we might as well fix it :)